controller/volume_controller.go

```diff
@@ -2995,16 +2996,27 @@ func (vc *VolumeController) processMigration(v *longhorn.Volume, es map[string]*
 		es[migrationEngine.Name] = migrationEngine
 	}
 
-	currentReplicas := map[string]*longhorn.Replica{}
+	currentAvailableReplicas := map[string]*longhorn.Replica{}
 	migrationReplicas := map[string]*longhorn.Replica{}
 	unknownReplicas := map[string]*longhorn.Replica{}
 	for _, r := range rs {
-		dataPath := types.GetReplicaDataPath(r.Spec.DiskPath, r.Spec.DataDirectoryName)
-		if r.Spec.FailedAt != "" {
+		isUnavailable, err := vc.IsReplicaUnavailable(r)
+		if err != nil {
+			return err
+		}
+		if isUnavailable {
 			continue
 		}
+		dataPath := types.GetReplicaDataPath(r.Spec.DiskPath, r.Spec.DataDirectoryName)
 		if r.Spec.EngineName == currentEngine.Name {
-			currentReplicas[dataPath] = r
+			if currentEngine.Status.ReplicaModeMap[r.Name] == types.ReplicaModeWO {
```

Comment:
So we still block while a rebuild is in progress, is that desired?

Reply:
Right. Launching the new migration replica process or adding it to the engine process will fail if the rebuilding is in progress. And I don't want to modify the longhorn-engine part, which will make things complicated.